src/dataflow/lib/iot.ts

```diff
@@ -203,12 +213,35 @@ export class IoT {
     const  { hubStore } = this.stores;
     const hub = hubStore.getHubById(hubId);
     const time = message.time ? parseFloat(message.time) * 1000 : Date.now();
+    let hubStoreSensors = 0;
     if (hub) {
+      hub.setHubUpdateTime(Date.now());
+      let messageChannels = 0;
+      let unmatchedChannel = false;
       for (const key in message) {
         if (message.hasOwnProperty(key) && key !== "time") {
-          hub.setHubChannelValue(key, String(message[key]));
-          hub.setHubChannelTime(key, time);
+          messageChannels++;
+          if (hub.getHubChannel(key)) {
+            hub.setHubChannelValue(key, String(message[key]));
+            hub.setHubChannelTime(key, time);
+          } else {
+            unmatchedChannel = true;
+          }
+        }
+      }
+
+      // WTD simulator doesn't currently return relay value in sensors message
+      // until fixed, need to handle both the hub and hub sim cases
+      hub.hubChannels.forEach(ch => {
+        if (ch.type !== "relay") {
+          hubStoreSensors++;
         }
+      });
```

Comment:
Note that at present the hub and the hub simulator behave differently.  The simulator does not send the relay status in the sensors MQTT message (while the hub does).  Therefore if we want to compare the device list in our store to the device list in the sensors message in hopes of detecting mismatches, we need to handle both the hub and the hub sim cases.  For now this means comparing the sensors in the hub store to the devices in the message (not the full list of devices in the hub store).  I requested a change to the hub-sim to unify the sensors message content, but in the meantime, we need to count sensor channels and do the more convoluted check below.